File: services/server/src/main/java/com/yaytsa/server/infra/persistence/repository/ImageRepository.java
package com.yaytsa.server.infra.persistence.repository;

import com.yaytsa.server.infra.persistence.entity.ImageEntity;
import com.yaytsa.server.infra.persistence.entity.ImageType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface ImageRepository extends JpaRepository<ImageEntity, UUID> {
    Optional<ImageEntity> findFirstByItemIdAndType(UUID itemId, ImageType type);
    List<ImageEntity> findAllByItemId(UUID itemId);
}

File: infra-persistence/library/src/main/kotlin/dev/yaytsa/persistence/library/repository/ImageRepository.kt
package dev.yaytsa.persistence.library.repository

import dev.yaytsa.persistence.library.entity.ImageJpa
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface ImageRepository : JpaRepository<ImageJpa, UUID> {
    fun findByEntityIdAndIsPrimaryTrue(entityId: UUID): ImageJpa?

    fun findByEntityId(entityId: UUID): List<ImageJpa>

    fun findByEntityIdInAndIsPrimaryTrue(entityIds: Collection<UUID>): List<ImageJpa>
}
